core/src/test/java/com/linecorp/armeria/server/RoutingContextTest.java

```diff
@@ -106,8 +106,9 @@ void testEquals() {
         ctx2 = new DefaultRoutingContext(virtualHost, "example.com",
                                          RequestHeaders.of(HttpMethod.GET, "/hello"), "/hello", "a=1", false);
 
-        assertThat(ctx1.hashCode()).isEqualTo(ctx2.hashCode());
-        assertThat(ctx1).isEqualTo(ctx2);
+        // Queries are different.
+        assertThat(ctx1.hashCode()).isNotEqualTo(ctx2.hashCode());
+        assertThat(ctx1).isNotEqualTo(ctx2);
```

Comment:
Could you add an equal comparison for `ctx1.param = "a=1&b=1"` and `ctx2.param = "a=1&b=1"`? 😉